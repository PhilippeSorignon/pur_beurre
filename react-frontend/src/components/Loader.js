import React from "react";

class Loader extends React.Component {
  render() {
    return (
      <div className="lds-ring">
        <div />
        <div />
        <div />
        <div />
      </div>
    );
  }
}

export default Loader;
