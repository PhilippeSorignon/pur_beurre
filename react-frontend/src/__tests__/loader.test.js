import React from 'react';
import ReactDOM from 'react-dom';
import Loader from '../components/Loader';

import { cleanup } from '@testing-library/react';
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Loader renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loader />, div);
});

it("Loader matches snapshot", () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
})
