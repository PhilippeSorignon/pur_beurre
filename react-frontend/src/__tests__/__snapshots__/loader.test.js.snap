// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<div
  className="lds-ring"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
